ui/packages/upload/src/utils.ts

```diff
@@ -15,7 +15,11 @@ export function normalise_file(
 			normalise_file(x, root);
 		}
 	} else if (file.is_file) {
-		file.data = root + "file=" + file.name;
+		if (root == null) {
```

Comment:
I think this is making it so that the `proxy=` route is always used even if you are not loading an external space?

The `zip_files` demo errors locally and on spaces for me and I think it's because the download link includes the `proxy` prefix

![image](https://user-images.githubusercontent.com/41651716/217027403-6387520c-b690-4eca-bb6c-57722553c716.png)

Reply:
I think it's because `root` is an empty string in this case (not undefined). When running in `dev` mode (`pnpm build`) the root is `localhost:7860` so that also tries to go through the `proxy` route but that doesn't work.

Reply:
Good catch @freddyaboulton can confirm the issue